src/BCFKeyValueData.cc

```diff
@@ -874,6 +892,28 @@ static Status validate_bcf(BCFBucketRange& rangeHelper, const bcf_hdr_t *hdr,
                                std::to_string(prev_pos) + " >= " + range(bcf).str());
     }
 
+    // verify record does not go over the length of the contig
+    const string &contig_name = contigs[bcf->rid].first;
+    size_t contig_len = contigs[bcf->rid].second;
+    if (bcf->pos + bcf->rlen > contig_len) {
+        return Status::Invalid("gVCF record is longer than contig ",
+                               range(bcf).str() + " " + to_string(contig_len) + " " + contig_name);
+    }
+
+    std::regex e ("[ACGTN]+");
+    for (int i=0; i < bcf->n_allele; i++) {
+        const string allele_i(bcf->d.allele[i]);
+        if (allele_i[0] == '<') {
+            // This is most likely a <NON_REF> symbol, or equivalent
+            continue;
+        }
+        if (regex_match(allele_i ,dnaRegExp))
+            continue;
+
+        return Status::Invalid("allele is not a recognized DNA sequence ",
+                               allele_i +  " " + range(bcf).str() + " " + contig_name);
```

Comment:
If you call `range(bcf).str(contigs) it will write out the contig name for you. https://github.com/dnanexus-rnd/GLnexus/blob/orodeh_gvcf_ingest_validation/include/types.h#L136